core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/domain/lookup/LookUpStrategyResolver.java

```diff
@@ -80,9 +80,8 @@ public LookUpStrategyResolver(DescriptorPolicy descriptorPolicy, LookUpStrategyT
      * @return never null
      */
     public LookUpStrategy determineLookUpStrategy(Object object) {
-        Class<?> objectClass = object.getClass();
-        return decisionCache.computeIfAbsent(objectClass, key -> {
-            if (object.getClass().isEnum()) {
+        return decisionCache.computeIfAbsent(object.getClass(), objectClass -> {
+            if (objectClass.isEnum()) {
```

Comment:
These changes reduce allocations in some cases by 5GB(!) per minute.

Reply:
The reason is because the difference between doing this code 5 billion times:

```
computeIfAbsent(a.b, k -> doStuff(a.b))
```

or this code 5 billion times:

```
computeIfAbsent(a.b, k -> doStuff(k))
```

The latter creates 1 lambda allocation, the former creates 5 billion lambda allocations.